### ISSUE TITLE
Add force_destroy option to aws_athena_workgroup

### ISSUE THREAD

**Issue by spartako** [terraform-providers/terraform-provider-aws#10085]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
Currently, Terraform fails when trying to delete a non-empty Athena Workgroup. This is good, however there are cases when we actually want to destroy the Workgroup, including everything within it. AWS CLI currently supports this use case.
Adding a `force_destroy` arg to the `aws_athena_workgroup` resource is useful to this use case, same as with `aws_s3_bucket` or `aws_athena_database`.

### New or Affected Resource(s)

* aws_athena_workgroup

### Potential Terraform Configuration

```hcl
resource "aws_s3_bucket" "results" {
  bucket = "athena-results-${lower(var.environment_name)}-${random_string.bucket_results_suffix.result}"
  force_destroy = true
}

resource "random_string" "bucket_results_suffix" {
  length = 6
  special = false
  upper = false
  number = false
}

resource "aws_athena_workgroup" "workgroup" {
  name = "${var.environment_name}_workgroup"
  configuration {
    enforce_workgroup_configuration    = true
    result_configuration {
      output_location = "s3://${aws_s3_bucket.results.bucket}/output/"
    }
  }
  force_destroy = true
}

resource "aws_athena_database" "database" {
  name   = "db_${lower(var.environment_name)}"
  bucket = aws_s3_bucket.results.bucket
  force_destroy = true
}
```

### References

aws-cli can destroy non empty workgroups: https://docs.aws.amazon.com/cli/latest/reference/athena/delete-work-group.html

**Comment by JoelClemence** [comment]
It's worth noting that having implemented the workaround and up versioned to latest terraform now gets the warning:
```
Warning: External references from destroy provisioners are deprecated

  on athena/workgroup/main.tf line 32, in resource "null_resource" "athena-cleanup":
  32:     command = <<-EOF
  33:       aws athena delete-work-group \
  34:         --work-group ${aws_athena_workgroup.athena-workgroup.name} \
  35:         --region ${var.region-name} \
  36:         --recursive-delete-option
  38:       sleep 5
  39:       EOF
```

**Comment by bflad** [comment]
Support for a `force_destroy` argument in the `aws_athena_workgroup` resource has been merged and will release with version 2.55.0 of the Terraform AWS Provider, later this week. Thanks to @leppikallio and @teraken0509 for their efforts on the implementation. 👍
